core/src/main/java/org/apache/accumulo/core/data/ArrayByteSequence.java

```diff
@@ -154,6 +154,15 @@ public int offset() {
     return offset;
   }
 
+  /**
+   * Reset the backing array for this byte sequence object. This is useful for object re-use.
+   */
+  public void reset(byte[] data) {
+    this.data = data;
+    this.offset = 0;
+    this.length = data.length;
```

Comment:
Could make this a bit more generic.

```suggestion
  public void reset(byte[] data. int offset, length) {
    this.data = data;
    this.offset = offset;
    this.length = length;
```

Reply:
Addressed in a0cc0d7